### PR TITLE
Fix BatchNormalization

### PR DESCRIPTION
**Pull request by abhaikollara**
Fixed picking of the normalized form corresponding to the training phase

from
```python
        return K.in_train_phase(normed,
                                normalize_in_training,
                                training=training)
```
to
```python
        return K.in_train_phase(normalize_in_training,
                                normed,
                                training=training)
```

### CODE REVIEW

File: keras/layers/normalization.py [hunk 1]
```diff
@@ -137,52 +137,48 @@ def call(self, inputs, training=None):
             inputs, self.gamma, self.beta, reduction_axes,
             epsilon=self.epsilon)
 
-        if training in {0, False}:
```
Comment (fchollet) [inline keras/layers/normalization.py]: We still need this part, for efficiency when the learning phase is static.

Comment (abhaikollara) [inline keras/layers/normalization.py]: I've undone that change

**Comment by fchollet** [comment]
FYI this broke batchnorm, but it's largely my mistake: the function `normalize_in_training` was misnamed.
